pkg/process/checks/system_info_windows.go

```diff
@@ -22,35 +21,39 @@ import (
 // the process messages.
 func CollectSystemInfo() (*model.SystemInfo, error) {
 	hi := platform.CollectInfo()
-	cpuInfo, err := cpu.GetCpuInfo()
+	cpuInfo := cpu.CollectInfo()
+	mi, err := winutil.VirtualMemory()
 	if err != nil {
 		return nil, err
 	}
-	mi, err := winutil.VirtualMemory()
+	physCount, err := cpuInfo.CPUPkgs.Value()
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("gohai cpuInfo.CPUPkgs: %w", err)
 	}
-	physCount, _ := strconv.ParseInt(cpuInfo["cpu_pkgs"], 10, 64)
+
 	// logicalcount will be the total number of logical processors on the system
 	// i.e. physCount * coreCount * 1 if not HT CPU
 	//      physCount * coreCount * 2 if an HT CPU.
-	logicalCount, _ := strconv.ParseInt(cpuInfo["cpu_logical_processors"], 10, 64)
+	logicalCount, _ := cpuInfo.CPULogicalProcessors.Value()
```

Comment:
I'm curious, why do we check for an error for `cpuInfo.CPUPkgs.Value()` but not `cpuInfo.CPULogicalProcessors.Value()`?

Reply:
I kept the previous behavior which is to ignore errors except specifically for the `CPUPkgs` field, because there is a division by that value.
The previous implementation assumed that `cpu.GetCpuInfo()` would return an error if any error occurs (in particular when collecting `CPUPkgs`) but the map would actually contain the default value if it failed to collect the field.
I replaced that error checking with the one of `CpuPkgs`.

Feel free to tell me if you think that function should return an error if any of the field fails to be collected.

Reply:
I think that makes sense, thank you!